test/components/figure_test.rb

```diff
@@ -0,0 +1,32 @@
+require 'component_test_helper'
+
+class FigureTest < ComponentTestCase
+  def component_name
+    "figure"
+  end
+
+  test "fails to render a figure when no data is given" do
+    assert_raise do
+      render_component({})
+    end
+  end
+
+  test "fails to render a figure when a href is provided but alt text isn't" do
```

Comment:
Reluctantly, yes – for now. The schema allow images without alt attributes: https://github.com/alphagov/govuk-content-schemas/blob/master/formats/definitions.json#L202

We might be able to add some logging to see how many images are being rendered without an alt attribute.